### PR TITLE
fixed resp.statusCode >= 400

### PR DESCRIPTION
**Pull request by EvanCarroll**

### CODE REVIEW

File: core.js [hunk 1]
```diff
@@ -110,18 +110,25 @@ module.exports = function (xhr) {
...
-                      body = JSON.parse(body);
-                  } catch (err) {
-                      if (options.error) return options.error(resp, 'error', err.message);
-                  }
-              }
-              if (options.success) return options.success(body, 'success', resp);
+          }
+          // XHR does not return error message >= 400.
+          else if ( resp.statusCode >= 400 ) {
+              if ( options.error ) return options.error(resp, 'error', '[ampersand-model] HTTP Status Code 4xx: Client Error' );
```
Comment (naugtur) [inline core.js]: This is actually incorrect. statusCode>=400 includes 5XX, so we can't return that message.

File: core.js [hunk 2]
```diff
@@ -110,18 +110,25 @@ module.exports = function (xhr) {
       // Make the request. The callback executes functions that are compatible
       // With jQuery.ajax's syntax.
       var request = options.xhr = options.xhrImplementation(ajaxSettings, function (err, resp, body) {
-          if (err || resp.statusCode >= 400) {
+          if (err) {
+              // Not sure when err !== 'undefined' && body
               if (options.error) return options.error(resp, 'error', err.message || body);
```
Comment (naugtur) [inline core.js]: This line is actually addressing #52 by returning the body if there was no clientside error. The only problem here is that err is null, so this line is the only one that needs to be fixed. I'll switch to commenting in #50 now.

**Comment by EvanCarroll** [comment]
Seems a lot cleaner in the patch. The body is also null -- so you need to generate the error as I did in the patch.

**Comment by naugtur** [comment]
if both error and body are null, it means there's no information about the error... 
In that case it seems you actually want a feature that'd return a message with the statusCode. It can be done, let's talk about it and see if others see the point of doing that.

**Comment by EvanCarroll** [comment]
Right, clearly that's what I want. But, moreover, it's needed and that's what this request does. Your just failing silently rather than failing because of bad conditional. That's a very small step up. The problem is only the ampersand-sync knows that the resp >= 400, even though we're triggering an error for it. That makes it very difficult to test and rather useless.

**Comment by naugtur** [comment]
current ampersand-sync behavior in xhr2-and-node does not differ from latest released ampersand-sync  as far as I know. 
and nothing is failing silently (failing silently would be not calling the error callback at all)

returning status code is a good idea, but this branch is not about changing the behavior.

**Comment by EvanCarroll** [comment]
So how do you get the reason for the failure? Failing silently is precisely invoking the error callback without a reason.. That's what we're doing here.

File: core.js (hunk 1 above)
Comment (EvanCarroll) [inline core.js]: Good point @naugtur , I can fix that but it sounds like you'll just reject the patch?

**Comment by EvanCarroll** [comment]
Force-pushed to reflect change in names, and the resp.statusCode oversight you pointed out.

**Comment by naugtur** [comment]
too much verbosity and unnecessary changes, force push and other merges. This PR is now too hard to merge.
I addressed the issue in #55

**Pull request closed without merging**